File: prakash/17_PalindromeCheck/PalindromeCheck.test.js
const isPalindrome = require("./PalindromeCheck");

describe("isPalindrome", () => {
  // Test 1: Valid palindrome containing spaces and punctuation
  test("Test with a valid palindrome containing spaces and punctuation", () => {
    const input = "A man, a plan, a canal: Panama";
    expect(isPalindrome(input)).toBe(true);
  });

  // Test 2: Non-palindrome
  test("Test with a non-palindrome", () => {
    const input = "hello";
    expect(isPalindrome(input)).toBe(false);
  });

  // Test 3: Empty string
  test("Test with an empty string", () => {
    const input = "";
    expect(isPalindrome(input)).toBe(true); // An empty string is considered a palindrome
  });

  // Test 4: String with only non-alphanumeric characters
  test("Test with a string containing only non-alphanumeric characters", () => {
    const input = "!!@@##";
    expect(isPalindrome(input)).toBe(true); // An empty string after removing non-alphanumeric characters
  });
});

File: Bipin/017_palindrome_check/palindrome_check_test.go
package main

import "testing"

// TestIsPalindrome tests the IsPalindrome function.
func TestIsPalindrome(t *testing.T) {
	// Define test cases
	tests := []struct {
		input    string
		expected bool
	}{
		{"A man, a plan, a canal: Panama", true}, // Test with a valid palindrome containing spaces and punctuation
		{"race a car", false},                    // Test with a non-palindrome
		{"", true},                               // Test with an empty string
		{"!!@@##$$", true},                       // Test with a string containing only non-alphanumeric characters
	}

	// Iterate through each test case
	for _, test := range tests {
		t.Run(test.input, func(t *testing.T) {
			result := IsPalindrome(test.input)
			if result != test.expected {
				t.Errorf("For input '%s', expected %t but got %t", test.input, test.expected, result)
			}
		})
	}
}
